### PR TITLE
mbedtls: some enhancement for PSA crypto core

### PR DESCRIPTION
**Pull request by valeriosetti**
- Enable legacy `CONFIG_MBEDTLS_CIPHER_AES_ENABLED` when `CONFIG_PSA_WANT_KEY_TYPE_AES` is set and Mbed TLS is the PSA crypto API provider (i.e. `CONFIG_MBEDTLS_PSA_CRYPTO_C`). This mimic what happens in Mbed TLS at build time.
- Enable `CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG` instead of `CONFIG_MBEDTLS_PSA_CRYPTO_LEGACY_RNG` if the platform has any CSPRNG enabled. This helps reducing  RAM/ROM footprint.

### CODE REVIEW

**Review by ceolin — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Comment by dkalowsk** [comment]
As this has a migration guide for 4.0 change, I'm guessing this is desired to be in 4.0.  Given that we're at RC2 now, it's going to need a GitHub Issue to make that happen.  I'm 90% sure this can just be linked to the deprecation of TinyCrypt but I'll leave that up to you :-)  If you think this should be part of 4.0 please add the Milestone.

**Comment by valeriosetti** [comment]
> As this has a migration guide for 4.0 change, I'm guessing this is desired to be in 4.0. Given that we're at RC2 now, it's going to need a GitHub Issue to make that happen. I'm 90% sure this can just be linked to the deprecation of TinyCrypt but I'll leave that up to you :-) If you think this should be part of 4.0 please add the Milestone.

This is only required for #79931 which is still drafted, so no this won't be part of 4.0. I'll move the documentation

**Comment by dkalowsk** [comment]
As discussed in the Release meeting today, we want to consider the deprecation of Tinycrypt a release blocker item for 4.0. That would mean Part1, Part2, and the mbedTLS changes would all need to be merged in.

@ceolin any objection to this course of action? The point was raised that Tinycrypt has been "just about to be deprecated" for 5 years now, but never seems to make it. We'd like to make it happen now.

File: tests/crypto/mbedtls_psa/testcase.yaml
Comment (str4t0m) [inline tests/crypto/mbedtls_psa/testcase.yaml:36]: Shouldn't it be: when there is **no** CSPRNG available in the last part of this comment?

Comment (valeriosetti) [reply tests/crypto/mbedtls_psa/testcase.yaml:36]: You're absolutely right. Thanks

**Review by jhedberg — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

File: doc/releases/migration-guide-4.0.rst
Comment (tomi-font) [inline doc/releases/migration-guide-4.0.rst:80]:
```suggestion
  is set), the Kconfig option :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG`
  is now the default choice for the random number source instead of
```

**Review by tomi-font — DISMISSED** [review]

**Comment by tomi-font** [comment]
please review @ceolin

**Review by str4t0m — DISMISSED** [review]

**Review by frkv — APPROVED** [review]
Good enhancements! Hope this gets in quickly

**Review by d3zd3z — DISMISSED** [review]

**Review by jhedberg — APPROVED** [review]

File: doc/releases/migration-guide-4.1.rst
Comment (tomi-font) [inline doc/releases/migration-guide-4.1.rst:35]:
```suggestion
  is now the default choice for random number source instead of
```

**Review by tomi-font — APPROVED** [review]
Minor nit I just realized. Thanks for the double rebase!

File: doc/releases/migration-guide-4.1.rst
Comment (tomi-font) [inline doc/releases/migration-guide-4.1.rst:37]:
```suggestion
  the ROM/RAM footprint of the Mbed TLS library.
```

**Comment by valeriosetti** [comment]
@ceolin now that 4.0 is out, the CI of this PR is green and we have some approval, can we please merge it?
This would reduce the dependency chain for #79931

**Comment by tomi-font** [comment]
> @ceolin now that 4.0 is out, the CI of this PR is green and we have some approval, can we please merge it? This would reduce the dependency chain for #79931

As per the rules the assignee (so @ceolin) needs to approve the PR for it to be mergeable.

**Review by ceolin — APPROVED** [review]

**Pull request merged**
